functions/__git.default_branch.fish

```diff
@@ -1,7 +1,10 @@
 function __git.default_branch -d "Fallback to main when master branch is not found"
   # Silently fail when not in git directory
   command git rev-parse --git-dir &>/dev/null; or return
-  if command git show-ref -q --verify refs/heads/master
+  set defaultBranch (command git config --get init.defaultBranch)
```

Comment:
I think we'll want to use `-l` (local) on this set, given the user might have a wider scoped defaultBranch variable defined, which we would then mangle.

> 
> • Variables may be explicitly set to universal, global or local. Variables with the  same
>   name in different scopes will not be changed.
> 
> • If  a  variable  is not explicitly set to be either universal, global or local, but has
>   been previously defined, the previous variable scope is used.

Reply:
pushed a new revision :)